drivers/timer/nrf_rtc_timer.c

```diff
@@ -63,7 +61,10 @@ void rtc1_nrf_isr(void *arg)
 	if (!IS_ENABLED(CONFIG_TICKLESS_KERNEL)) {
 		u32_t next = last_count + CYC_PER_TICK;
 
-		if (counter_sub(next, t) < MIN_DELAY) {
+		/* As below: we're guaranteed to get an interrupt as
```

Comment:
This could happen to the previous implementation too, it's nothing new.  And it's fundamental with any kind of comparator architecture like this.  If you can arbitrarily preempt the timer driver it's always possible to mess up its logic.

Stated more simply: if you need good realtime timing, you can't be preempting the driver that provides that.

But one good workaround for an app that aboslutely had to use a ZLI or NMI (or something at a higher priority than timer) would be to chain to a lower priority exception (like PendSV) that can then flag a timer interrupt synchronously, which will reset the state.  You could have a tiny high priority ("MetaIRQ") thread for exactly this purpose.  With a little work you could optimize this to recognize if a timer ISR had been interrupted and skip the extra step.